src/cargo/ops/cargo_test.rs

```diff
@@ -114,15 +119,20 @@ fn run_doc_tests(options: &TestOptions,
                  -> CargoResult<Vec<ProcessError>> {
     let mut errors = Vec::new();
     let config = options.compile_opts.config;
-    let libs = compilation.package.targets().iter()
-                    .filter(|t| t.doctested())
-                    .map(|t| (t.src_path(), t.name(), t.crate_name()));
-    for (lib, name, crate_name) in libs {
+
+    let mut libs = vec![];
```

Comment:
You may be able to avoid the intermediate vec here (mostly for clarity) via a `flat_map` instead of a vec + extend